### ISSUE TITLE
How to run with local model in local folder?

### ISSUE THREAD

**Issue by FieldMarshallVague** [microsoft/guidance#324]
Hi, 

I'm pretty new to python and AI, but I've been running local models for both LLMs and SD successfully.  

I am trying to figure out how to load one of my LLM models from a local folder (even if it's just copying the model to a folder in the guidance project), but I can't see any examples of this and it looks like it's either downloading the models or using remote APIs to execute the programs.  Is this the case?  Does it, or can it work with locally-installed models?  If not, is this a planned feature?  

Maybe I've completely missed the point, or some examples, I don't know, but I'm very tired! :)

To be clear, I want to use a model file (e.g. Llama 7b) I've downloaded previously, I want to use MS Guidance on that model without downloading it again. 

Thanks.

**Comment by FieldMarshallVague** [comment]
Oh, wow, I missed the uppercase T in the model loading line.

**Comment by itsPreto** [comment]
@FieldMarshallVague how did you get it working? I'm also a bit confused on this-- I've got a bunch of models already dowloaded but it's not clear how to work with those directly

**Comment by FieldMarshallVague** [comment]
In the docs, there's an example for local models.  You have to use your local path (I copied them to a sub-folder of the guidance project I created).  

I incorrectly tried this first (I'm using powershell):
```
vicuna = guidance.llms.transformers(".\models\vicuna_13B", device_map="auto")
```

But that gave me the error above.  Then I realised I wasn't using the correct method of llms (which is uppercase T).

```
llama = guidance.llms.Transformers(".\models\llama-7b", device=0)
```

That caused another error about too much recursion, which turned out to be an issue with the model I was using.   I think I was using a GGML model or something.  I'm sorry, I don't remember.

**Comment by itsPreto** [comment]
@FieldMarshallVague wow i thought the transformers constructor was for fetching/downloading from HF!

I'll try this out thank you!

**Comment by itsPreto** [comment]
@FieldMarshallVague Do I need to clone the repo and do the same thing you did? I'm trying to just pass in the absolute path to the model and its still trying to download the HF model--
<img width="1372" alt="Screenshot 2023-07-24 at 9 09 32 PM" src="https://github.com/microsoft/guidance/assets/45348368/3058a00e-13a1-4986-88ba-5f9b1bd276ea">

**Comment by FieldMarshallVague** [comment]
@itsPreto yeah, that's what I thought.  But it definitely supports local files.  You just have to figure out the right syntax.  I'm on windows, looks like you're on Linux?  I also found that the hyphens in the name of the model folder were causing problems.  No idea why, seems like a bug (possibly because it's using HF validator even on the local files?).   I ended up renaming mine to URL-safe names.

But to be clear:  Mine is still not working properly (as mentioned above).  I stopped when I saw this repo hadn't received any recent updates and I have no idea where to start debugging this project.  I also looked at LMQL, btw.  I'd rather use Guidance for their straightforward syntax (LMQL is a horrible syntax), but I'm really not sure what's going on with this repo.  I couldn't find much info on it.
